onetl/connection/db_connection/kafka.py

```diff
@@ -212,3 +225,53 @@ def _validate_auth(cls, values: dict) -> dict:  # noqa: N805
             "Please provide either `keytab` and `user`, or `password` and "
             "`user` for Kerberos auth, or none of parameters for anonymous auth",
         )
+
+    @staticmethod
+    def _calculate_hash(value: str):
+        hash_object = hashlib.md5()  # noqa: S303
+        hash_object.update(value.encode("utf-8"))
+        return hash_object.hexdigest()
+
+    def _get_jaas_conf(self):
+        service_name = Kafka._calculate_hash(  # noqa: WPS437
+            f"{self.addresses}{self.user}{self.password}{self.cluster}",
+        )
+        if self.password is not None:
+            return self._password_jaas(service_name)
+
+        return self._keytab_jaas(service_name)
+
+    def _keytab_jaas(self, service_name) -> str:
+        if self.keytab is None:
+            raise ValueError("keytab path is None")
+
+        if self.deploy_keytab:
+            keytab = self._move_keytab()
+            self.spark.sparkContext.addFile(str(keytab))
+        else:
+            keytab = self.keytab
+
+        return (
+            "com.sun.security.auth.module.Krb5LoginModule required\n"
+            f'keyTab="{keytab}"\n'
+            f'principal="{self.user}"\n'
+            f'serviceName="{service_name}"\n'
+            "renewTicket=true\n"
+            "storeKey=true\n"
+            "useKeyTab=true\n"
+            "useTicketCache=false;"
+        )
+
+    def _password_jaas(self, service_name) -> str:
+        return (
+            "org.apache.kafka.common.security.plain.PlainLoginModule required\n"
+            f'serviceName="{service_name}"\n'
+            f'username="{self.user}"\n'
+            f'password="{self.password}";'
+        )
+
+    def _move_keytab(self) -> LocalPath:
```

Comment:
```suggestion
    def _move_keytab_to_cwd(self) -> LocalPath:
```
А чтобы не игнорировать `self.keytab`, потому что он может быть `None`, просто сделай его аргументом метода